src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Design/build/netcoreapp1.0/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Design.targets

```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <MvcRazorContentRoot Condition="'$(MvcRazorContentRoot)'==''">$(MSBuildProjectDirectory)</MvcRazorContentRoot>
+    <MvcRazorOutputPath Condition="'$(MvcRazorOutputPath)'==''">$(OutputPath)</MvcRazorOutputPath>
+  </PropertyGroup>
+  <Target Name="Precompile"
+    AfterTargets="Precompile-Build;Precompile-Publish">
+    <ItemGroup>
+      <ExecArgs Include="
+        --runtimeconfig $(OutputPath)$(MSBuildProjectName).runtimeconfig.json;
+        --depsfile $(OutputPath)$(MSBuildProjectName).deps.json;
+        $(MSBuildThisFileDirectory)Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.Design.dll;
+        $(MSBuildProjectDirectory);
+        --output-path=$(MvcRazorOutputPath);
+        --application-name=$(MSBuildProjectName);
+        --content-root=$(MvcRazorContentRoot);" />
+      <ExecArgs Condition="$(DelaySign) == true"
+        Include="--delay-sign" />
+    </ItemGroup>
+    <Exec Command="dotnet exec @(ExecArgs->'%(Identity)', ' ')" />
+  </Target>
+  <Target Name="Precompile-Build"
```

Comment:
These shoudl be conditional. Right now, it's probably running the tool twice, once during build and once during publish.